File: project/core/lib/external/kafka/src/main/java/vdi/core/kafka/EventSource.java
package vdi.core.kafka;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum EventSource {
  OBJECT_STORE,
  FULL_RECONCILER,
  SLIM_RECONCILER,
  UPDATE_META_LANE,
  INSTALL_DATA_LANE,
  ;

  @JsonCreator
  public static EventSource fromString(String value) {
    for (var e : EventSource.values())
      if (e.name().equals(value))
        return e;

    throw new IllegalArgumentException("unrecognized EventSource value: $value");
  }
}

File: project/core/lib/external/kafka/src/main/kotlin/vdi/core/kafka/EventSource.kt
package vdi.core.kafka

import com.fasterxml.jackson.annotation.JsonCreator

enum class EventSource {
  ObjectStore,
  FullReconciler,
  SlimReconciler,
  UpdateMetaLane,
  InstallDataLane,
  ;

  companion object {
    @JvmStatic
    @JsonCreator
    fun fromString(value: String): EventSource {
      for (enum in entries)
        if (enum.name == value)
          return enum

      throw IllegalArgumentException("unrecognized EventSource value: $value")
    }
  }
}
